.github/PULL_REQUEST_TEMPLATE.md

```diff
@@ -0,0 +1,50 @@
+## Pull request checklist
+
+Please check if your PR fulfills the following requirements:
+
+- [ ] Tests for the changes have been added (for bug fixes / features)
+- [ ] Docs have been reviewed and added / updated if needed (for bug fixes / features)
+- [ ] Build (`npm run build`) was run locally for affected output targets
+
+## Pull request type
+
+<!-- Please do not submit updates to dependencies unless it fixes an issue. -->
+
+<!-- Please try to limit your pull request to one type, submit multiple pull requests if needed. -->
+
+Please check the type of change your PR introduces:
+
+- [ ] Bugfix
+- [ ] Feature
+- [ ] Code style update (formatting, renaming)
+- [ ] Refactoring (no functional changes, no api changes)
+- [ ] Build related changes
+- [ ] Documentation content changes
+- [ ] Other (please describe):
+
+## What is the current behavior?
+
+<!-- Please describe the current behavior that you are modifying. -->
+
+<!-- Issues are required for both bug fixes and features. -->
```

Comment:
Do we want to make this a requirement? I can see this adding additional noise to the repo where folks add an issue just for the sake of filling out this field

Reply:
In Framework, we do this to discourage PRs without a verified/triaged issue. This prevents individuals opening up a PR that does not align with our goals.

We also require a reproduction for issues or a valid use case for a feature, so we have something to validate the PR against. 

It also aids when individuals run into similar problems, they can search the existing issues to find a match vs. PRs do not show up in the issue search.

Reply:
I agree with the latter two points here:

> We also require a reproduction for issues or a valid use case for a feature, so we have something to validate the PR against.

⬆️ this one in particular

> It also aids when individuals run into similar problems, they can search the existing issues to find a match vs. PRs do not show up in the issue search.

But I'm not sure I follow the first point

> In Framework, we do this to discourage PRs without a verified/triaged issue. This prevents individuals opening up a PR that does not align with our goals.


Is the discouragement coming from having to open an issue in the first place? IE what's to stop me from putting a PR together, realizing I "need" an issue, add one, then say I met the minimum requirements?  Have y'all found that to be enough?

Reply:
I think if someone does post an issue just to say they meet the requirements, it makes for a better spot to put discussion on why their issue isn't a good fit, ask for clarification, etc. since other people will be more likely to find it and/or chime in. The author will also be forced to justify their PR more thoroughly by filling out the issue template as well.

Reply:
Do we feel aligned on this stance? We can always adapt this template as we receive feedback from the community.

Reply:
Yeah, I think we're good.  I'm OK with moving forward with this as-is - if we feel we need to, we can adjust the template like @sean-perkins said